code/modules/mining/drilling/difficulties.dm

```diff
@@ -0,0 +1,151 @@
+//Cave difficulty levels store most stats that affect cavegen based on seismic level.area
+//Golems also store a reference to the current difficulty, currently used only for golem_ore_mult
+
+/datum/cave_difficulty_level //using a datum here lets me simplify the spawn pool logic and declutter cave_generator.dm in the process
+	var/level //number value is easier to work with in some situations
+
+	var/golem_ore_mult = 1
+	var/vein_ore_mult = 1
+
+	//number of golems to spawn *per squad*
+	var/golem_count_melee = 0
+	var/golem_count_ranged = 0
+	var/golem_count_mixed = 0 // the "mixed" pool is the concatenation of the melee and ranged pools
+
+	//lists of types and weights
+	var/list/golem_weights_melee = list()
+	var/list/golem_weights_ranged = list()
+
+/datum/cave_difficulty_level/proc/get_golem_spawns()
+	var/list/golems_to_spawn = list() //list of golem types to spawn
+	if(golem_count_melee)
+		for(var/c = 0, c < golem_count_melee, c++)
+			golems_to_spawn += pickweight(golem_weights_melee)
```

Comment:
`for(var/c in 0 to golem_count_melee)` style is noticeably more performant, or so other coders say.
Probably some kind of optimization on BYOND side.
Or rather lack of optimization when C-style loops are being converted from DM.